app/controllers/authentication.js

```diff
@@ -55,165 +56,242 @@ var Controller = function(){
+    User.findOne({ email: req.body.email }, '+password', function (err, user) {
       if (!user) {
         return res.status(401).send({ message: 'Invalid email and/or password' });
       }
-      user.comparePassword(req.body.password, function(err, isMatch) {
+      user.comparePassword(req.body.password, function (err, isMatch) {
         if (!isMatch) {
           return res.status(401).send({ message: 'Invalid email and/or password' });
         }
         res.send({ token: createJWT(user) });
       });
     });
-  }
-  this.logout = function(req, res){
+  };
+
+  this.logout = function (req, res) {
      req.logout();
-     res.json({logout: 'success'});
-  }
-  var loginRequiredResponse = function(req, res){
-    res.status(404).json({error: 'login required'});
-  }
-  var apiKeyRequiredResponse = function(req, res){
-    res.status(404).json({error: 'apikey required'});
-  }
-  this.loginRequired = function(req, res, next){
-    if( !req.user){
+     res.json({ logout: 'success' });
+  };
+
+  var loginRequiredResponse = function (req, res) {
+    res.status(404).json({ error: 'login required' });
+  };
+
+  var apiKeyRequiredResponse = function (req, res) {
+    res.status(404).json({ error: 'apikey required' });
+  };
+
+  this.loginRequired = function (req, res, next) {
+    if (!req.user) {
       loginRequiredResponse(req, res);
     } else {
-      next()
+      next();
     }
-  }
+  };
 
-  this.apiKeyRequired = function(req, res, next){
-    User.apiKeyExists(req.query.apiKey, function(error, ok){
-      if(error){
+  this.apiKeyRequired = function (req, res, next) {
+    User.apiKeyExists(req.query.apiKey, function (error, ok) {
+      if (error) {
         apiKeyRequiredResponse(req, res);
-      } else if(!ok){
+      } else if (!ok) {
         apiKeyRequiredResponse(req, res);
       } else {
         next();
       }
-    })
-  }
-  this.adminRequired = function(req, res, next){
-    if( !req.user ){
+    });
+  };
+
+  this.adminRequired = function (req, res, next) {
+    if (!req.user) {
       loginRequiredResponse(req, res);
-    } else if(!req.user.account_level){
-      res.status(404).json({error: 'admin required'});
+    } else if (!req.user.account_level) {
+      res.status(404).json({ error: 'admin required' });
     } else {
       next();
     }
-  }
+  };
+
   /*
    |--------------------------------------------------------------------------
    | Login with GitHub
    |--------------------------------------------------------------------------
    */
-  this.github = function(req, res) {
+  this.github = function (req, res) {
     var accessTokenUrl = 'https://github.com/login/oauth/access_token';
     var userApiUrl = 'https://api.github.com/user';
-    
-    var params = {
-      code: req.body.code,
-      client_id: req.body.clientId,
-      client_secret: nconf.get('github').clientSecret,
-      redirect_uri: req.body.redirectUri
+
+    var authorization = function (callback) {
```

Comment:
add more logs here to help understand if something goes wrong.. some debug traces in every waterfall steps..